updated ClipboardItem and Clipboard documentation and examples using …

tayloregivens opened the pull request:
…supports()


Updated the web API ClipboardItem and Clipboard pages to describe the supports() method. I also updated the code samples that write data to the clipboard so they use the supports method as well.

My team at Microsoft created the supports() static method and so we wanted to go ahead and document on MDN as well.

Commit:
updated ClipboardItem and Clipboard documentation and examples using supports()

tayloregivens committed:
Update files/en-us/web/api/clipboard/write/index.md

fixing quotations based on linter suggestion

Co-authored-by: github-actions[bot] <41898282+github-actions[bot]@users.noreply.github.com>

tayloregivens committed:
Update files/en-us/web/api/clipboard/write/index.md

fixing spacing based on linter suggestion

Co-authored-by: github-actions[bot] <41898282+github-actions[bot]@users.noreply.github.com>

tayloregivens committed:
Update files/en-us/web/api/clipboarditem/supports/index.md

Co-authored-by: github-actions[bot] <41898282+github-actions[bot]@users.noreply.github.com>

tayloregivens committed:
Update files/en-us/web/api/clipboarditem/supports/index.md

Co-authored-by: github-actions[bot] <41898282+github-actions[bot]@users.noreply.github.com>

tayloregivens committed:
Update files/en-us/web/api/clipboard/writetext/index.md

Co-authored-by: github-actions[bot] <41898282+github-actions[bot]@users.noreply.github.com>

tayloregivens committed:
Update files/en-us/web/api/clipboard/write/index.md

Co-authored-by: github-actions[bot] <41898282+github-actions[bot]@users.noreply.github.com>

tayloregivens committed:
Update files/en-us/web/api/clipboarditem/clipboarditem/index.md

Co-authored-by: github-actions[bot] <41898282+github-actions[bot]@users.noreply.github.com>

snianu commented on files/en-us/web/api/clipboard/write/index.md:
````diff
@@ -54,14 +54,18 @@ Note that for this particular case, you could just as readily use `Clipboard.wri
 ```
 
-The `setClipboard()` method begins by creating a new a {{domxref("Blob")}} object.
+The `setClipboard()` method begins with an if statement that returns true or false based on whether the clipboard supports 'text/plain' MIME type. If the clipboard does support the MIME type, then a new {{domxref("Blob")}} object is created.
````
Nit: Can we make it a little more succinct? e.g.
`The `setClipboard()` method first checks for the support of `text/plain` MIME type. If it's supported, a new {{domxref("Blob")}} object is created.`
Please feel free to rephrase this to make it more concise.

wbamberg replied:
I would say something like:

```suggestion
The `setClipboard()` method first uses the {{domxref("ClipboardItem.supports()")}} static method to check whether the clipboard supports the `"text/plain"` MIME type.

If the clipboard does support plain text, then `setClipboard()` creates a new {{domxref("Blob")}} object.
```

But as Patrick said, is it good practice to use `supports()` for plain text?

snianu commented on files/en-us/web/api/clipboard/write/index.md:
```diff
@@ -81,12 +85,16 @@ const canvas = document.getElementById("canvas");
 canvas.addEventListener("click", copyCanvasContentsToClipboard);
 
 async function copyCanvasContentsToClipboard() {
```
Can we change the below sentence in ln 80 to describe the changes we made in this example?
> This example writes the canvas to a blob, using the default MIME type of `image/png`, and then writes the blob to the clipboard.

Here we can mention that the author doesn't have to generate the data for images if the `image/png` type isn't supported by the browser.

snianu commented on files/en-us/web/api/clipboarditem/index.md:
```diff
@@ -27,6 +27,13 @@ _This interface provides the following properties._
+_This interface defines the following methods._
+
+- {{domxref("ClipboardItem.supports", "supports()")}}
+  - : Returns true or false whether a {{Glossary("MIME type")}} type is supported. This allows us to detect if a {{Glossary("MIME type")}} is supported by the clipboard before attempting to write data.
```
Nit: _before attempting to generate the data for the {{Glossary("MIME type")}} and write to the clipboard_.

wbamberg replied:
```suggestion
  - : Checks whether a given {{Glossary("MIME type")}} is supported by the clipboard. This allows us to detect if a MIME type is supported by the clipboard before attempting to write data.
```

(although I would personally omit the second sentence and have it in the main page)

snianu commented on files/en-us/web/api/clipboarditem/index.md:
```diff
@@ -38,22 +45,25 @@ _This interface defines the following methods._
 ### Writing to the clipboard
 
-Here we're writing a new {{domxref("ClipboardItem.ClipboardItem", "ClipboardItem()")}} to the system clipboard by requesting a PNG image using the {{domxref("Fetch API")}}, and in turn, the {{domxref("Response.blob()", "responses' blob()")}} method, to create the new `ClipboardItem`.
+Here we're writing a new {{domxref("ClipboardItem.ClipboardItem", "ClipboardItem()")}} to the system clipboard by requesting a PNG image using the {{domxref("Fetch API")}}, and in turn, the {{domxref("Response.blob()", "responses' blob()")}} method, to create the new `ClipboardItem`. Then we use {{domxref("ClipboardItem.supports", "supports()")}} to determine if the MIME data type is supported be ...
```
Looks like we changed it to first check for the support of the PNG image MIME type and _then_ generate the data and write to the clipboard, right? Can we change this text to reflect the changes we made here?

snianu commented on files/en-us/web/api/clipboarditem/supports/index.md:
```diff
@@ -0,0 +1,65 @@
+
+## Specifications
+
+{{Specifications}}
```
Spec link for `supports`: https://w3c.github.io/clipboard-apis/#dom-clipboarditem-supports

captainbrosset replied:
This link should come from BCD. The `{{Specifications}}` macro will pull it in automatically if it's present in BCD (browser-compat-data).

captainbrosset replied:
Just verified, the data is present: https://github.com/mdn/browser-compat-data/blob/main/api/ClipboardItem.json#L175C24-L175C88

snianu reviewed (commented):
Added some comments. PTAL, thanks!

wbamberg committed:
Apply Reviewdog suggestions

Co-authored-by: github-actions[bot] <41898282+github-actions[bot]@users.noreply.github.com>

captainbrosset commented on files/en-us/web/api/clipboarditem/supports/index.md:
```diff
@@ -0,0 +1,65 @@
+short-title: supports()
+slug: Web/API/ClipboardItem/supports
+page-type: web-api-static-method
+browser-compat: api.ClipboardItem.supports
```
I think this should be:

```suggestion
browser-compat: api.ClipboardItem.supports_static
```

to match the BCD data: https://github.com/mdn/browser-compat-data/blob/main/api/ClipboardItem.json#L173

captainbrosset commented on files/en-us/web/api/clipboard/writetext/index.md:
```diff
@@ -45,7 +45,11 @@ button.addEventListener("click", () => writeClipboardText("<empty clipboard>"));
 async function writeClipboardText(text) {
   try {
-    await navigator.clipboard.writeText(text);
+    if (ClipboardItem.supports("text/plain")) {
```
I wonder if we really need to change examples that deal with plain text. Are there cases where plain text is **not** supported?

wbamberg replied:
If I read the spec right, no: https://w3c.github.io/clipboard-apis/#mandatory-data-types.

wbamberg commented on files/en-us/web/api/clipboarditem/supports/index.md:
```diff
@@ -0,0 +1,65 @@
+
+## Examples
+
+In the following example, we're writing data to the clipboard via the {{domxref("clipboard.write()")}} method. Then we utilize {{domxref("ClipboardItem.supports", "supports()")}} to see if the {{Glossary("MIME type")}} is supported by the clipboard before attempting to write it to the clipboard. Using {{domxref("ClipboardItem.supports", "supports()")}} saves us from costly and unesseary CPU cycle ...
```
```suggestion
### Writing an image to the clipboard

In the following example, we're writing an image to to the clipboard.

We use `supports()` to check whether the `"image/png"` MIME type is supported by the clipboard before fetching the image and writing it using {{domxref("clipboard.write()")}}.

We also use a [`try..catch`](/en-US/docs/Web/JavaScript/Reference/Statements/try...catch) statement to catch any errors while writing the data to the clipboard.
```
 
- use a descriptive title for the example, and a sentence giving an overview of what we are doing
- don't self-link with `{{domxref("ClipboardItem.supports", "supports()")}}`
- avoid passive voice in the bit about `try...catch`, and link to the JS docs

I wasn't sure about "Using {{domxref("ClipboardItem.supports", "supports()")}} saves us from costly and unnecessary CPU cycles from attempting to write unsupported formats to the clipboard." I think as a rule examples should just illustrate, that is they "show not tell" so things about the motivation for the method should live in the method general description. But I'm not sure this is needed there either, perhaps it is too obvious?

wbamberg commented on files/en-us/web/api/clipboarditem/supports/index.md:
````diff
@@ -0,0 +1,65 @@
+In the following example, we're writing data to the clipboard via the {{domxref("clipboard.write()")}} method. Then we utilize {{domxref("ClipboardItem.supports", "supports()")}} to see if the {{Glossary("MIME type")}} is supported by the clipboard before attempting to write it to the clipboard. Using {{domxref("ClipboardItem.supports", "supports()")}} saves us from costly and unesseary CPU cycle ...
+
+```js
+async function writeClipImg() {
````
```suggestion
async function writeImageToClipboard() {
```

...this is strictly optional but I like more readable method names.

wbamberg replied:
Oh I see the other examples use `writeClipImg` as well. So I guess this is even more optional :).

wbamberg commented on files/en-us/web/api/clipboard/write/index.md:
```diff
@@ -54,14 +54,18 @@ Note that for this particular case, you could just as readily use `Clipboard.wri
-  const blob = new Blob([text], { type });
-  const data = [new ClipboardItem({ [type]: blob })];
-  await navigator.clipboard.write(data);
+  if (ClipboardItem.supports("text/plain")) {
+    const type = "text/plain";
```
It's a bit weird to use a literal for `text/plain`, then declare a variable for it in the next line. Perhaps:

```suggestion
  const type = "text/plain";
  if (ClipboardItem.supports(type)) {
```

?

wbamberg reviewed (dismissed):
Thank you for the PR, @tayloregivens !

wbamberg commented:
Oh, also I missed that the directory for `supports` should have the `_static` suffix. See for example `json_static` or `error_static` in https://github.com/mdn/content/tree/main/files/en-us/web/api/response. We need to do this to support static and instance methods that have the same name, like https://developer.mozilla.org/en-US/docs/Web/API/Response/json_static and https://developer.mozilla.org/en-US/docs/Web/API/Response/json.

tayloregivens committed:
Update files/en-us/web/api/clipboarditem/supports/index.md

Co-authored-by: wbamberg <will@bootbonnet.ca>

tayloregivens committed:
Update files/en-us/web/api/clipboarditem/supports/index.md

Co-authored-by: wbamberg <will@bootbonnet.ca>

tayloregivens committed:
Update files/en-us/web/api/clipboarditem/supports/index.md

Co-authored-by: wbamberg <will@bootbonnet.ca>

tayloregivens committed:
Update files/en-us/web/api/clipboarditem/supports/index.md

Co-authored-by: wbamberg <will@bootbonnet.ca>

tayloregivens committed:
Update files/en-us/web/api/clipboarditem/supports/index.md

Co-authored-by: wbamberg <will@bootbonnet.ca>

tayloregivens committed:
Update files/en-us/web/api/clipboard/write/index.md

Co-authored-by: wbamberg <will@bootbonnet.ca>

tayloregivens committed:
Update files/en-us/web/api/clipboard/write/index.md

Co-authored-by: wbamberg <will@bootbonnet.ca>

tayloregivens committed:
Update files/en-us/web/api/clipboarditem/supports/index.md

Co-authored-by: Patrick Brosset <patrickbrosset@gmail.com>

tayloregivens committed:
Update files/en-us/web/api/clipboarditem/index.md

Co-authored-by: wbamberg <will@bootbonnet.ca>

tayloregivens committed:
Update index.md clipboard item example description

Updated "Writing to the clipboard" example description to be more accurate.

tayloregivens committed:
Creating supports_static folder

tayloregivens committed:
Update index.md in supports_static

tayloregivens committed:
Delete files/en-us/web/api/clipboarditem/supports directory

wbamberg commented on files/en-us/web/api/clipboarditem/supports_static/index.md:
```diff
@@ -0,0 +1,71 @@
+---
+title: "ClipboardItem: supports() static method"
+short-title: supports()
+slug: Web/API/ClipboardItem/supports
```
```suggestion
slug: Web/API/ClipboardItem/supports_static
```

...hopefully this will fix the CI error.

wbamberg committed:
Update files/en-us/web/api/clipboarditem/supports_static/index.md

wbamberg commented on files/en-us/web/api/clipboard/write/index.md:
````diff
@@ -55,13 +55,19 @@ button.addEventListener("click", () => setClipboard("<empty clipboard>"));
 ```
 
-The `setClipboard()` method begins by creating a new a {{domxref("Blob")}} object.
+The `setClipboard()` method first uses the {{domxref("ClipboardItem.supports()")}} static method to check whether the clipboard supports the `"text/plain"` MIME type.
````
```suggestion
The `setClipboard()` method first uses the {{domxref("ClipboardItem.supports_static", "ClipboardItem.supports()")}} static method to check whether the clipboard supports the `"text/plain"` MIME type.
```

(domxref is not very nice to use with statics. Hopefully this ought to fix the flaws reported for this page)

wbamberg commented on files/en-us/web/api/clipboarditem/index.md:
```diff
@@ -27,6 +27,13 @@ _This interface provides the following properties._
+
+_This interface defines the following methods._
+
+- {{domxref("ClipboardItem.supports", "supports()")}}
```
```suggestion
- {{domxref("ClipboardItem.supports_static", "ClipboardItem.supports()")}} 
```

wbamberg commented on files/en-us/web/api/clipboarditem/index.md:
```diff
@@ -38,21 +45,24 @@ _This interface defines the following methods._
 ### Writing to the clipboard
 
-Here we're writing a new {{domxref("ClipboardItem.ClipboardItem", "ClipboardItem()")}} to the system clipboard by requesting a PNG image using the {{domxref("Fetch API")}}, and in turn, the {{domxref("Response.blob()", "responses' blob()")}} method, to create the new `ClipboardItem`.
+Here we use {{domxref("ClipboardItem.supports", "supports()")}} to determine if 'img/png' MIME data type is supported before generating the image data and writing it to the clipboard. The image is generated using the {{domxref("Fetch API")}}, and in turn, the {{domxref("Response.blob()", "responses' blob()")}} method is used to create a `ClipboardItem` that is written to the clipboard.
```
```suggestion
Here we use {{domxref("ClipboardItem.supports", "supports()")}} to determine if 'img/png' MIME data type is supported before generating the image data and writing it to the clipboard. The image is generated using the {{domxref("Fetch API")}}, and in turn, the {{domxref("Response.blob()", "responses' blob()")}} method is used to create a `ClipboardItem` that is written to the clipboard.
```
```suggestion
Here we use {{domxref("ClipboardItem.supports_static", "supports()")}}  to determine if 'img/png' MIME data type is supported before generating the image data and writing it to the clipboard. The image is generated using the {{domxref("Fetch API")}}, and in turn, the {{domxref("Response.blob()", "responses' blob()")}} method is used to create a `ClipboardItem` that is written to the clipboard.
```

wbamberg committed:
Fix domxref calls to work with static

captainbrosset commented on files/en-us/web/api/clipboard/write/index.md:
```diff
@@ -55,13 +55,19 @@ button.addEventListener("click", () => setClipboard("<empty clipboard>"));
-  const blob = new Blob([text], { type });
-  const data = [new ClipboardItem({ [type]: blob })];
-  await navigator.clipboard.write(data);
+  if (ClipboardItem.supports(type)) {
```
As discussed offline, let's remove this change since `text/plain` is always supported.

hamishwillee replied:
1. We should list the types in supports() as per https://github.com/mdn/content/pull/33262#discussion_r1597862947 - the lists are small and discrete, while the current implication of that doc is you can use any MIME type.
2. I don't understand it yet, but the interesting optional type I'd like here is the custom type prefixed by `web `.

FYI, I'm showing an interest because this is being supported in FF127 and making sure there are docs is obviously part of that https://github.com/mdn/content/issues/33564 . @tayloregivens Great so see work already happening on this!

captainbrosset commented on files/en-us/web/api/clipboard/write/index.md:
````diff
@@ -55,13 +55,19 @@ button.addEventListener("click", () => setClipboard("<empty clipboard>"));
 ```
 
-The `setClipboard()` method begins by creating a new a {{domxref("Blob")}} object.
+The `setClipboard()` method first uses the {{domxref("ClipboardItem.supports_static", "ClipboardItem.supports()")}} static method to check whether the clipboard supports the `"text/plain"` MIME type.
````
And revert this change too, to match the code block above.

captainbrosset commented on files/en-us/web/api/clipboarditem/index.md:
```diff
@@ -27,6 +27,13 @@ _This interface provides the following properties._
+_This interface defines the following methods._
+
+- {{domxref("ClipboardItem.supports_static", "ClipboardItem.supports()")}}
+  - : Checks whether a given {{Glossary("MIME type")}} is supported by the clipboard. This allows us to detect if a MIME type is supported by the clipboard before attempting to write data.
```
```suggestion
  - : Checks whether a given {{Glossary("MIME type")}} is supported by the clipboard. This allows to detect if a MIME type is supported by the clipboard before attempting to write data.
```

The usage of "us" felt a bit inconsistent with other reference docs.

captainbrosset replied:
I see this page uses "us"/"we" in the examples section. I think it's fine there because it's more of a tutorial. But in the reference part, it feels a bit out of place.

wbamberg replied:
I agree about "us" but "allows to detect" is not quite right. "enables to website to detect" is better if a bit longwinded.

captainbrosset reviewed (approved):
Drive-by approval with a couple of comments. Let's wait for Will to do a more official review.

wbamberg commented on files/en-us/web/api/clipboarditem/index.md:
```diff
@@ -38,21 +45,24 @@ _This interface defines the following methods._
 ### Writing to the clipboard
 
-Here we're writing a new {{domxref("ClipboardItem.ClipboardItem", "ClipboardItem()")}} to the system clipboard by requesting a PNG image using the {{domxref("Fetch API")}}, and in turn, the {{domxref("Response.blob()", "responses' blob()")}} method, to create the new `ClipboardItem`.
+Here we use {{domxref("ClipboardItem.supports_static", "supports()")}} to determine if 'img/png' MIME data type is supported before generating the image data and writing it to the clipboard. The image is generated using the {{domxref("Fetch API")}}, and in turn, the {{domxref("Response.blob()", "responses' blob()")}} method is used to create a `ClipboardItem` that is written to the clipboard.
```
```suggestion
Here we use {{domxref("ClipboardItem.supports_static", "supports()")}} to check whether the 'img/png' MIME data type is supported. If it is, we fetch the image with the ["Fetch API"](/en-US/docs/Web/API/Fetch_API), and then read it into a {{domxref("Blob")}}, which we can use to create a `ClipboardItem` that is written to the clipboard.
```

(however, if we decide not to use `supports()` with "img/png", per https://github.com/mdn/content/pull/33262#discussion_r1585203697, then obviously this won't apply, and we should just revert this part of the PR.)

hamishwillee replied:
Ah, this the same problem I see with https://github.com/mdn/content/pull/33262/files#r1597871059 ?  I guess it is OK to for one example to use a mandatory type, but we should have at least one example that does not and ideally cross link.

hamishwillee replied:
WillB suggestion updated with `image/` rather than `img`, and changed to new type.
I've decided that in this particular case we'll add a note after to indicate that this check isn't strictly necessary.

```suggestion
Here we use {{domxref("ClipboardItem.supports_static", "supports()")}} to check whether the 'image/png' MIME data type is supported. If it is, we fetch the image with the ["Fetch API"](/en-US/docs/Web/API/Fetch_API), and then read it into a {{domxref("Blob")}}, which we can use to create a `ClipboardItem` that is written to the clipboard.
```

wbamberg reviewed (commented):
Thank you for the updates @tayloregivens !

I had just one more suggested change, but I think we also have an open question about whether it's appropriate to use `supports()` with "img/png" (https://github.com/mdn/content/pull/33262#discussion_r1585203697). If not we should use a different MIME type for the `supports()` page and remove the `supports()` check from the examples that use "img/png" as well as the plain text ones.

tayloregivens committed:
Revert changes to writeText file

Since plain text is always supported no need to add support() static method to example

tayloregivens committed:
changes reverted to write file

text always support so no need for supports()

tayloregivens committed:
removed "us" in clipboardItem index.md

hamishwillee commented on files/en-us/web/api/clipboarditem/supports_static/index.md:
```diff
@@ -0,0 +1,71 @@
+### Parameters
+
+- `type`
+  - : A string representing a {{Glossary("MIME type")}}.
```
The spec is not quite this simple. 
- It defines [mandatory types](https://w3c.github.io/clipboard-apis/#mandatory-data-types) and [optional types](https://w3c.github.io/clipboard-apis/#optional-data-types-x).
- For the optional types it defines custom types which are prefixed with `web ` and have a second part that passes a mime-type-format check.

Given the set is small and it isn't just any MIME type, we should list them here

hamishwillee commented on files/en-us/web/api/clipboard/write/index.md:
```diff
@@ -73,20 +73,24 @@ A `try..catch` block could be used to catch any errors writing the data.
 ### Write canvas contents to the clipboard
 
-This example writes the canvas to a blob, using the default MIME type of `image/png`, and then writes the blob to the clipboard.
+This example writes the canvas to a blob, using the default MIME type of `image/png`, and then writes the blob to the clipboard. Generating image data is avoided if 'image/png' isn't supported by the browser.
```
This feels a bit more natural?

```suggestion
This example first checks if the browser allows 'image/png' data to be written to the clipboard.
If so, it writes the canvas to a blob, using the default MIME type of `image/png`, and then writes the blob to the clipboard. 
```

Given this is a mandatory type the method has to return true right? Seems a bit weird to even bother testing it since if the method is defined it will return true. I'd be tempted to use an optional type for this.

hamishwillee replied:
PS Given this doesn't rely on network, could we perhaps make this a live example?

hamishwillee commented:
@tayloregivens There are also some merge conflicts here. Let me know if there is anything I can do to help you progress this.

hamishwillee committed:
Update files/en-us/web/api/clipboard/write/index.md

Co-authored-by: github-actions[bot] <41898282+github-actions[bot]@users.noreply.github.com>

hamishwillee committed:
Fix write() example

hamishwillee commented:
Hi @tayloregivens 
I've fixed up the example/conflicts and rebased (this is now a live example). I'll continue with necessary updates tomorrow, unless you have time to look at them.

hamishwillee committed:
supports() - list supported types

hamishwillee commented on files/en-us/web/api/clipboarditem/supports_static/index.md:
```diff
@@ -0,0 +1,83 @@
+
+### Writing an image to the clipboard
+
+The following example fetches an SVG image to a blob, and then writes it to the clipboard.
```
FYI, trivially changed supports to `"image/svg+xml"`. Using `supports()` with image/png should always return true, and is pointless if we know we're always working with PNG. So unless we update this so the example doesn't know what file and hence MIME type it is writing to clipboard in advance, this is better.

hamishwillee committed:
Update files/en-us/web/api/clipboarditem/index.md

hamishwillee committed:
index - Change to svg

hamishwillee committed:
Note on clipboard.write() about not needing the check

hamishwillee commented:
@tayloregivens I have fixed this up to address the comments. Let's see if we can get it through review!

@wbamberg Can you please look at this again. The main changes are:
1. The `ClipboardItem` and `supports()` page examples use `image/svg+xml` for the mime types.
2. `Clipboard.write()` still uses `image/png` and is a live example (merged in from main and slightly modified). I've added a note to make it clear that the `supports()` check is not really necessary, but highlight some cases where it could be (i.e. showing "for information").
3. The `supports()` method now states the supported MIME types. Note that it omits one in the specification that no one has implemented.

tayloregivens commented:
> @tayloregivens I have fixed this up to address the comments. Let's see if we can get it through review!
> 
> @wbamberg Can you please look at this again. The main changes are:
> 
> 1. The `ClipboardItem` and `supports()` page examples use `image/svg+xml` for the mime types.
> 2. `Clipboard.write()` still uses `image/png` and is a live example (merged in from main and slightly modified). I've added a note to make it clear that the `supports()` check is not really necessary, but highlight some cases where it could be (i.e. showing "for information").
> 3. The `supports()` method now states the supported MIME types. Note that it omits one in the specification that no one has implemented.

Thanks for addressing the remaining comments and updating the 'supports()' page to use 'image/svg+xml' and making a note that supports() isn't necessary with 'image/png' on the Clipboard.write()

for 3. the supports method allows web authors to specify any arbitrary type and returns false for types that aren't supported by the browser. If you did want to add extra information there you could say that the mandatory types, html, text, and png are always supported and don't require using supports() static method.

hamishwillee commented on files/en-us/web/api/clipboarditem/supports_static/index.md:
```diff
@@ -0,0 +1,83 @@
+
+{{APIRef("Clipboard API")}} {{securecontext_header}}
+
+The **`supports()`** static method of the {{domxref("ClipboardItem")}} interface indicates whether or not the given {{Glossary("MIME type")}} is supported by the clipboard.
+
```
```suggestion
The **`supports()`** static method of the {{domxref("ClipboardItem")}} interface returns `true` if the given {{Glossary("MIME type")}} is supported by the clipboard, and `false` otherwise.

Note that the [Clipboard API](/en-US/docs/Web/API/Clipboard_API) mandates support for plain text, HTML and PNG files.
The `supports()`  method will always return `true` for these MIME types, so testing them is unnecessary .

```

hamishwillee committed:
Update files/en-us/web/api/clipboarditem/supports_static/index.md

wbamberg committed:
Update files/en-us/web/api/clipboarditem/supports_static/index.md

Co-authored-by: github-actions[bot] <41898282+github-actions[bot]@users.noreply.github.com>

wbamberg committed:
Merge branch 'main' into clipboard-supports

wbamberg reviewed (approved):
👍 thank you @tayloregivens and @hamishwillee for your work on this PR!
